Update and extend lazer mod articles (#14794)
ppy/osu-wiki: wiki/Gameplay/Game_modifier/Perfect_(lazer)/en.md

**Note:** For the full list of all [lazer](/wiki/Client/Release_stream/Lazer) mods, see [Game modifier (lazer)](/wiki/Gameplay/Game_modifier_(lazer))
:::

> The **Perfect** mod immediately triggers a failure if the player can no longer achieve an SS [grade](/wiki/Gameplay/Grade) on a [beatmap](/wiki/Beatmap). In other words, the player will automatically fail when getting anything worse than GREAT hit on a [hit circle](/wiki/Gameplay/Hit_object/Hit_circle), [sliderhead](/wiki/Gameplay/Hit_object/Slider/Sliderhead), [don or kat](/wiki/Gameplay/Hit_object/Hit), note or [hold note](/wiki/Beatmap/Pattern/osu!mania/Hold_note). The player will also fail when missing a [slider tick](/wiki/Gameplay/Hit_object/Slider/Slider_tick), [reverse arrow](/wiki/Gameplay/Hit_object/Slider/Repeat_slider), [slidertail](/wiki/Gameplay/Hit_object/Slider/Slidertail), [fruit](/wiki/Gameplay/Hit_object/Fruit), [hyperfruit](/wiki/Gameplay/Hit_object/Hyperfruit), [large droplet](/wiki/Gameplay/Hit_object/Juice_stream#drop) or [small droplet](/wiki/Gameplay/Hit_object/Juice_stream#droplet), but not when missing a [drumroll](/wiki/Gameplay/Hit_object/Drumroll), [swell](/wiki/Gameplay/Hit_object/Swell) or a [banana](/wiki/Gameplay/Hit_object/Banana).

Reviewer: ```suggestion
The **Perfect** mod immediately triggers a failure if the player can no longer achieve an SS [grade](/wiki/Gameplay/Grade) on a [beatmap](/wiki/Beatmap):

- When getting anything worse than GREAT on [hit circle](/wiki/Gameplay/Hit_object/Hit_circle), [sliderhead](/wiki/Gameplay/Hit_object/Slider/Sliderhead), [don or kat](/wiki/Gameplay/Hit_object/Hit), note or [hold note](/wiki/Beatmap/Pattern/osu!mania/Hold_note).
- When missing a [slider tick](/wiki/Gameplay/Hit_object/Slider/Slider_tick), [reverse arrow](/wiki/Gameplay/Hit_object/Slider/Repeat_slider), [slidertail](/wiki/Gameplay/Hit_object/Slider/Slidertail), [fruit](/wiki/Gameplay/Hit_object/Fruit), [hyperfruit](/wiki/Gameplay/Hit_object/Hyperfruit), [large droplet](/wiki/Gameplay/Hit_object/Juice_stream#drop) or [small droplet](/wiki/Gameplay/Hit_object/Juice_stream#droplet)

The objects that are safe to miss are [drumrolls](/wiki/Gameplay/Hit_object/Drumroll), [swells](/wiki/Gameplay/Hit_object/Swell) and [bananas](/wiki/Gameplay/Hit_object/Banana).
```
should make it more bearable to read

Reviewer: also, what about osu! spinners? guess they belong to the second list?

Reviewer: on a second thought, while I like the idea of the *completeness* here, this could be greatly simplified without the list:

```suggestion
The **Perfect** mod immediately triggers a failure if the player can no longer achieve an SS [grade](/wiki/Gameplay/Grade) on a [beatmap](/wiki/Beatmap), which will happen after missing or getting anything worse than GREAT on almost all types of hit objects. The objects that can be missed without fail are [drumrolls](/wiki/Gameplay/Hit_object/Drumroll), [swells](/wiki/Gameplay/Hit_object/Swell) and [bananas](/wiki/Gameplay/Hit_object/Banana).
```

## Customisation

![Perfect customisation options in the game client](/wiki/Gameplay/Game_modifier_(lazer)/img/customise/PF.png)

- **Require perfect hits** (default: disabled): Getting anything other than a PERFECT hit (rainbow-coloured 300) will result in a failure. *This option is only available in osu!mania.*
- **Restart on fail** (default: enabled): Automatically restart the map after a failure.

Changing any of these settings will **not** cause your scores to be unranked.

## Trivia

> - When using Perfect in combination with [Easy (EZ)](/wiki/Gameplay/Game_modifier/Easy_(lazer)), the extra lives are taken into account, allowing the player to get a few non-perfect hits before failing.

Reviewer: side note: I'm curious if this produces a rankable score -- will check later

Reviewer: incredibly, yes: https://osu.ppy.sh/scores/7181580451
(I have already overwritten it with a clear pass, but it would have given me 5 pp as indicated on the score page)

Author: Yep, for SD too. Some players even have those scores in their top plays: https://osu.ppy.sh/scores/6362579553

Reviewer: in relation to that, do you think the below could help emphasize this?
```suggestion
- When using Perfect in combination with [Easy (EZ)](/wiki/Gameplay/Game_modifier/Easy_(lazer)), the extra lives are taken into account, allowing the player to get a few non-perfect hits before failing and then possibly submit the score.
```
